src/star-id.cpp

```diff
@@ -503,4 +504,37 @@ StarIdentifiers PyramidStarIdAlgorithm::Go(
     return identified;
 }
 
+// for ordering Quaternions in tracking mode votes map
+bool operator<(const Quaternion& l, const Quaternion& r) {
```

Comment:
This is not an appropriate general-purpose comparison for quaternions, just write it as a separate function and pass as a custom comparator to std::sort

Also, `||` is your friend.